File: Algorithm/baekjoon/swift/BJ10815.swift
//10815
//10815
func binarySearch(values: [Int], findValue: Int, start: Int, end: Int) -> Bool{
    //swift의 특징 = 상수!!!
    var s = start
    var e = end
    //정렬을 안에서 하게 되면 느려짐
    while(s <= e){
        let mid = (s + e) / 2
        if values[mid] > findValue {
            e = mid - 1
        }else if values[mid] < findValue {
            s = mid + 1
        }else{
            return true
        }
    }
    return false
}

var n = readLine()
var cards = readLine()?.split(separator: " ")
    .map{Int($0)!}.sorted()
var m = readLine()
var values = readLine()?.split(separator: " ")
    .map{Int($0)!}

for value in values!{
    if binarySearch(values: cards!, findValue: value, start: 0, end: (cards?.count)!-1){
        print("1", terminator:" ")
    }else{
        print("0", terminator:" ")
    }
}

File: Algorithm/baekjoon/java/BJ10815.java
package baekjoon.java;

/**
 * Created by changmin on 2018. 1. 16..
 */
import java.util.*;
public class BJ10815 {
    public static void main(String[] args){
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int cards[] = new int[n];

        for(int i = 0; i < n; ++i){
            cards[i] = sc.nextInt();
        }

        int m = sc.nextInt();
        int values[] = new int[m];

        for(int i = 0; i < m; ++i){
            values[i] = sc.nextInt();
        }

        String output = findNumber(cards, values);
        System.out.println(output);

    }

    public static String findNumber(int[] n, int[] values){
        Arrays.sort(n);
        StringBuilder output = new StringBuilder();

        for(int value : values){
            if(binarySearch(n, value, 0, n.length-1)){
                output.append(1 + " ");
            }else{
                output.append(0 + " ");
            }
        }

        return output.toString();
    }
    static boolean binarySearch(int[] n, int value, int start, int end){
        while(start <= end){
            int mid = (start + end) / 2;
            if(n[mid] > value){
                end = mid - 1;
            }
            else if (n[mid] < value){
                start = mid + 1;
            }
            else{
                return true;
            }
        }
        return false;
    }
}
